libs/jwst-codec/src/doc/history.rs

```diff
@@ -0,0 +1,193 @@
+use std::collections::{HashMap, VecDeque};
+
+use serde::Serialize;
+
+use super::*;
+
+#[derive(Debug)]
+struct GreenTree(HashMap<Id, String>);
+
+impl GreenTree {
```

Comment:
add some comments about how does it work and some internal details?

Reply:
done